node/src/protocol/consensus.rs

```diff
@@ -160,6 +163,10 @@ impl ConsensusProtocol for StartedState {
                                 epoch,
                                 contract_state.old_epoch
                             );
+
```

Comment:
I know that the participants are not available in all states, but can we set them 1 time at the beginning of the advance function?

Reply:
I wonder if I can just generalize over the contract_state and set the participants like that, because there's some cases where the participants was never set, but doesn't seem like it's too much of an issue